src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/ADRController.kt

```diff
@@ -167,42 +167,41 @@ class ADRController(private val encryption: Encryption,
     }
 
     @PostMapping("/datasets/{id}/resource/{resourceType}/{modelCalibrateId}")
-    @Suppress("ReturnCount")
+    @Suppress("ReturnCount", "LongParameterList")
     fun pushFileToADR(@PathVariable id: String,
                       @PathVariable resourceType: String,
                       @PathVariable modelCalibrateId: String,
                       @RequestParam resourceFileName: String,
-                      @RequestParam resourceId: String?): ResponseEntity<String>
+                      @RequestParam resourceId: String?,
+                      @RequestParam description: String) : ResponseEntity<String>
```

Comment:
`description` is now a required parameter, which may be the cause of the `ADRTests` failure.
However, rather than passing the description it might make more sense to retrieve `uploadMetadata` using `HintrAPIClient`. We do this for `downloadSpectrum` and `downloadSummary`, and `getPlottingMetadata` already exists, so maybe `getUploadMetadata` should also be added?

Reply:
We spoke about getting the data from HintrAPIClient but resolved to having it in frontend instead because we will allow users to be able to edit/alter description on the dialog at some point.

Reply:
Ah very good point - thanks